### PR TITLE
Disable most VM tests

### PR DESCRIPTION
**Pull request by edolstra**
###### Motivation for this change

Unfortunately VM tests take an excessive amount of time and memory to evaluate (more than half an hour, dozens of gigabytes of RAM). The `tested` job alone takes 5 minutes and ~12 GB. So disable most VM tests. Most of these should be moved into separate repos/flakes anyway in the future.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by grahamc** [comment]
This is unfortunate, I think they are extremely valuable. I wonder if there is a way we can test each of these as part of the release, but not in the same evaluation? I hate to lose permanently such an important thing. None of the removals specifically made me squirm.

File: nixos/release-combined.nix [hunk 1]
```diff
@@ -66,12 +66,11 @@ in rec {
         (all nixos.tests.containers-ip)
         nixos.tests.chromium.x86_64-linux or []
         (all nixos.tests.firefox)
-        (all nixos.tests.firefox-esr)
+        #(all nixos.tests.firefox-esr)
         (all nixos.tests.firewall)
         (all nixos.tests.fontconfig-default-fonts)
         (all nixos.tests.gnome3-xorg)
         (all nixos.tests.gnome3)
-        (all nixos.tests.pantheon)
```
Comment (worldofpeace) [inline nixos/release-combined.nix]: :eyes: I could suggest better tests to remove than something I worked months on to ensure is working, and an environment that is actually rather popular.

Comment (grahamc) [reply nixos/release-combined.nix]: Given the complexity of the pantheon ecosystem and the fact that it is one test, I am +1 on putting pantheon back.

**Comment by dtzWill** [comment]
Perhaps conditionally disable? For a "quick-tests" vs "release-test-all-the-things" jobset?

If resources are needed, perhaps a fundraising round would be a good idea? (or is the sort of agility lost w/these not easily addressed that way?)

Perhaps there are ways to slim the eval impact of our vm tests, with some re-use or something? (not sure how much that's been looked at...)

Disabled tests seem like they might as well be removed, which would be unfortunate indeed.
Since most contributors don't have the ability to run these tests regularly, would pushing to flakes be useful for evaluation costs or because .. these tests aren't run anymore by NixOS infra?
(not sure if the suggestion is to push into flakes for some eval-time improvements that don't require removing them...?)

File: nixos/release-combined.nix (hunk 1 above)
Comment (worldofpeace) [reply nixos/release-combined.nix]: there's also gnome's installed tests which are highly helpful for us that are commented https://github.com/NixOS/nixpkgs/blob/master/nixos/tests/installed-tests/default.nix. But I do understand that's about 17 test cases.

**Comment by grahamc** [comment]
https://github.com/NixOS/hydra/issues/715

File: nixos/release-combined.nix [hunk 2]
```diff
@@ -97,11 +97,11 @@ in rec {
         (all nixos.tests.ipv6)
         (all nixos.tests.i3wm)
         (except ["aarch64-linux"] nixos.tests.keymap.azerty)
```
Comment (worldofpeace) [inline nixos/release-combined.nix]: azerty was commented before I re-added these https://github.com/NixOS/nixpkgs/commit/bf49181373643d3e2df1a06cb6e1efde1a2dec3e#diff-0f15ebe03b218d11d461288a9b051eeb

File: nixos/tests/all-tests.nix [hunk 3]
```diff
@@ -21,298 +21,298 @@ let
...
   nat.firewall = handleTest ./nat.nix { withFirewall = true; };
   nat.firewall-conntrack = handleTest ./nat.nix { withFirewall = true; withConntrackHelpers = true; };
   nat.standalone = handleTest ./nat.nix { withFirewall = false; };
-  ndppd = handleTest ./ndppd.nix {};
-  neo4j = handleTest ./neo4j.nix {};
-  nesting = handleTest ./nesting.nix {};
-  netdata = handleTest ./netdata.nix {};
+  #ndppd = handleTest ./ndppd.nix {};
+  #neo4j = handleTest ./neo4j.nix {};
+  #nesting = handleTest ./nesting.nix {};
```
Comment (worldofpeace) [inline nixos/tests/all-tests.nix]: this tests nesting.clone and nesting.children, IMHO not a great omission

File: nixos/tests/all-tests.nix [hunk 4]
```diff
@@ -21,298 +21,298 @@ let
...
-  os-prober = handleTestOn ["x86_64-linux"] ./os-prober.nix {};
-  osrm-backend = handleTest ./osrm-backend.nix {};
-  overlayfs = handleTest ./overlayfs.nix {};
-  packagekit = handleTest ./packagekit.nix {};
-  pam-oath-login = handleTest ./pam-oath-login.nix {};
-  pam-u2f = handleTest ./pam-u2f.nix {};
+  #openstack-image-userdata = (handleTestOn ["x86_64-linux"] ./openstack-image.nix {}).userdata or {};
+  #openstack-image-metadata = (handleTestOn ["x86_64-linux"] ./openstack-image.nix {}).metadata or {};
+  #orangefs = handleTest ./orangefs.nix {};
+  #os-prober = handleTestOn ["x86_64-linux"] ./os-prober.nix {};
```
Comment (worldofpeace) [inline nixos/tests/all-tests.nix]: it scares me to not run this continuously, but it's actually broken so nvm https://hydra.nixos.org/job/nixos/trunk-combined/nixos.tests.os-prober.x86_64-linux

File: nixos/tests/all-tests.nix [hunk 5]
```diff
@@ -21,298 +21,298 @@ let
...
-  keepalived = handleTest ./keepalived.nix {};
-  kerberos = handleTest ./kerberos/default.nix {};
-  kernel-latest = handleTest ./kernel-latest.nix {};
+  #jackett = handleTest ./jackett.nix {};
+  #jellyfin = handleTest ./jellyfin.nix {};
+  #jenkins = handleTest ./jenkins.nix {};
+  #kafka = handleTest ./kafka.nix {};
+  #keepalived = handleTest ./keepalived.nix {};
+  #kerberos = handleTest ./kerberos/default.nix {};
+  #kernel-latest = handleTest ./kernel-latest.nix {};
```
Comment (worldofpeace) [inline nixos/tests/all-tests.nix]: we have the latest in the releases so I think that's not the best omission either, testing being omitted is fine though.

**Review by worldofpeace — CHANGES_REQUESTED** [review]
These omissions made me uncomfortable.
But doing this certainly is :crying_cat_face:

File: nixos/tests/all-tests.nix (hunk 4 above)
Comment (worldofpeace) [reply nixos/tests/all-tests.nix]: https://github.com/NixOS/nixpkgs/issues/80041

**Comment by vcunat** [comment]
Note: as #79907 wasn't mentioned, I didn't notice and pushed ceb90b08ef8 (20.03 only for now).

File: nixos/tests/all-tests.nix [hunk 6]
```diff
@@ -21,298 +21,298 @@ let
...
-  ceph-multi-node = handleTestOn ["x86_64-linux"] ./ceph-multi-node.nix {};
-  certmgr = handleTest ./certmgr.nix {};
-  cfssl = handleTestOn ["x86_64-linux"] ./cfssl.nix {};
+  #borgbackup = handleTest ./borgbackup.nix {};
+  #buildbot = handleTest ./buildbot.nix {};
+  #caddy = handleTest ./caddy.nix {};
+  #cadvisor = handleTestOn ["x86_64-linux"] ./cadvisor.nix {};
+  #cassandra = handleTest ./cassandra.nix {};
+  #ceph-single-node = handleTestOn ["x86_64-linux"] ./ceph-single-node.nix {};
+  #ceph-multi-node = handleTestOn ["x86_64-linux"] ./ceph-multi-node.nix {};
```
Comment (devhell) [inline nixos/tests/all-tests.nix]: Ouch. Ceph without tests make me feel queasy.

**Comment by flokli** [comment]
I don't really think simply disabling tests is the right way forward here. Also, having test in `all-tests.nix` (and in `nixosTest.*`) makes them discoverable.

It surely might ease some of the memory consumption issues we're currently facing, but having those tests around, and having most of the "stable ones" included in the tested jobset, and in this ensuring the features described there don't suddenly break is a very nice property we should not give up.

I'd really prefer if we could teach hydra to evaluate parts of that individually, as proposed in https://github.com/NixOS/hydra/issues/715.

**Comment by bjornfor** [comment]
The NixOS VM tests is a major selling point for me. Removing (so many of) them due to ~12 GiB of memory use and CPU cycles? I don't get that. Human cycles are much more expensive, and a change like this will mean more human cycles will have to be spent.

**Comment by flokli** [comment]
With all the changes that happened in the meantime on master and 20.03, I assume we don't plan to disable mostly all tests anymore. Please reopen if I'm wrong.

**Pull request closed without merging**